### ISSUE TITLE
cuDNN batch norm operator needs to update to use cuDNN helper functions.

### ISSUE THREAD

**Issue by qingqing01** [PaddlePaddle/Paddle#5238]
The cuDNN batch norm operator uses the  cuDNN API directly, for example using [`platform::dynload::cudnnCreateTensorDescriptor`](
https://github.com/PaddlePaddle/Paddle/blob/develop/paddle/operators/batch_norm_op.cu#L69) and [`platform::dynload::cudnnDestroyTensorDescriptor`](https://github.com/PaddlePaddle/Paddle/blob/develop/paddle/operators/batch_norm_op.cu#L160) directly. But we have some cuDNN helper functions to create and destroy the cuDNN descriptor conveniently.  These cuDNN helper functions had been used in cuDNN convolution operators.
